js/app.js

```diff
@@ -1,123 +1,92 @@
+CryptoOptionsController.$inject = ['$scope', '$httpParamSerializer'];
+function CryptoOptionsController($scope, $httpParamSerializer) {
+    var active_cryptos_div = document.getElementById("active_cryptos");
+    var active_cryptos = JSON.parse(active_cryptos_div.dataset.active_cryptos);
+    $scope.no_display_error = true;
+    $scope.active_cryptos = active_cryptos;
+    $scope.crypto_selecter  = true;
+    //fetch url params
+    $scope.order_id = getParameterByNameBlocko("select_crypto");
+
+    $scope.checkout_order_url = function() {
+        var params = getParameterByNameBlocko('wc-api');
+        if (params)
+            params = {
+                "wc-api": params
+            };
         else
-            param = {};
-        var item = $resource(window.location.pathname, param);
-        return item;
-    });
-
-var flyp_base = 'https://flyp.me/api/v1';
-
-service.factory('AltcoinNew', function($resource) {
-    var rsc = $resource(flyp_base + '/order/new');
-    return rsc;
-});
-
-service.factory('AltcoinAccept', function($resource) {
-    var rsc = $resource(flyp_base + '/order/accept');
-    return rsc;
-});
-
-service.factory('AltcoinCheck', function($resource) {
-    var rsc = $resource(flyp_base + '/order/check');
-    return rsc;
-});
-
-service.factory('AltcoinInfo', function($resource) {
-    var rsc = $resource(flyp_base + '/order/info');
-    return rsc;
-});
-
-service.factory('AltcoinAddRefund', function($resource) {
-    var rsc = $resource(flyp_base + '/order/addrefund');
-    return rsc;
-});
-
-service.factory('AltcoinLimits', function($resource) {
-    var rsc = $resource(flyp_base + '/order/limits/:coin/BTC', {
-        coin: '@coin'
-    });
-    return rsc;
-});
-
-service.factory('WpAjax', function($resource) {
-    var rsc = $resource(ajax_object.ajax_url);
-    return rsc;
-});
-
-app = angular.module("shopping-cart-demo", ["monospaced.qrcode", "shoppingcart.services"]);
-
-
-app.config(function($compileProvider) {
-    $compileProvider.aHrefSanitizationWhitelist(/^\s*(https?|ftp|mailto|data|chrome-extension|bitcoin|ethereum|litecoin):/);
-    // Angular before v1.2 uses $compileProvider.urlSanitizationWhitelist(...)
-});
+            params = {};
+        params.show_order = $scope.order_id;
+        params.crypto = $scope.crypto.code;
+        var url = window.location.pathname;
+        var serializedParams = $httpParamSerializer(params);
+        if (serializedParams.length > 0) {
+            url += ((url.indexOf('?') === -1) ? '?' : '&') + serializedParams;
+        }
+        return url;
+    }
 
-function getParameterByNameBlocko(name, url) {
-    if (!url) {
-        url = window.location.href;
+    //Select Blockonomics crypto
+    $scope.select_blockonomics_crypto = function(blockonomics_crypto) {
+        $scope.crypto_selecter  = false;
+        $scope.spinner = true;
+        $scope.crypto = $scope.active_cryptos[blockonomics_crypto];
+        $scope.crypto.code = blockonomics_crypto;
```

Comment:
Can't find the use of $scope.crypto inside CryptoOptionsController.  After we refactor *checkout_order_url* we just pass it order_id and *blockonomics_crypto* to that function

Reply:
Fixed in cd0b822